pkg/scheduler/algorithm/predicates/metadata_test.go

```diff
@@ -507,3 +507,235 @@ func TestPredicateMetadata_ShallowCopy(t *testing.T) {
+					},
+				},
+			},
+			nodes: []*v1.Node{
+				{ObjectMeta: metav1.ObjectMeta{Name: "nodeA", Labels: map[string]string{"hostname": "nodeA"}}},
+			},
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "aaa-normal", Labels: map[string]string{"aaa": ""}},
+			},
+			wantAffinityPods:     nil,
+			wantAntiAffinityPods: nil,
+		},
+		{
+			name: "no pod has label violates with incoming pod's anti-affinity",
+			existingPods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "bbb-normal", Labels: map[string]string{"bbb": ""}},
+					Spec:       v1.PodSpec{NodeName: "nodeA"},
+				},
+			},
+			nodes: []*v1.Node{
+				{ObjectMeta: metav1.ObjectMeta{Name: "nodeA", Labels: map[string]string{"hostname": "nodeA"}}},
+			},
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "aaa-anti", Labels: map[string]string{"aaa": ""}},
+				Spec: v1.PodSpec{
+					Affinity: &v1.Affinity{
+						PodAntiAffinity: &v1.PodAntiAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: newPodAffinityTerms([]string{"aaa"}),
+						},
+					},
+				},
+			},
+			wantAffinityPods:     make(map[string][]*v1.Pod),
+			wantAntiAffinityPods: make(map[string][]*v1.Pod),
+		},
+		{
+			name: "existing pod has label violates with incoming pod's anti-affinity",
+			existingPods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "aaa-normal", Labels: map[string]string{"aaa": ""}},
+					Spec:       v1.PodSpec{NodeName: "nodeA"},
+				},
+			},
+			nodes: []*v1.Node{
+				{ObjectMeta: metav1.ObjectMeta{Name: "nodeA", Labels: map[string]string{"hostname": "nodeA"}}},
+			},
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "aaa-anti", Labels: map[string]string{"aaa": ""}},
+				Spec: v1.PodSpec{
+					Affinity: &v1.Affinity{
+						PodAntiAffinity: &v1.PodAntiAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: newPodAffinityTerms([]string{"aaa"}),
+						},
+					},
+				},
+			},
+			wantAffinityPods: make(map[string][]*v1.Pod),
+			wantAntiAffinityPods: map[string][]*v1.Pod{
+				"nodeA": {
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "aaa-normal", Labels: map[string]string{"aaa": ""}},
+						Spec:       v1.PodSpec{NodeName: "nodeA"},
+					},
+				},
+			},
+		},
+		{
+			name: "incoming pod's anti-affinity has more than one term - existing pod violates all terms",
+			existingPods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "normal", Labels: map[string]string{"aaa": "", "bbb": ""}},
+					Spec:       v1.PodSpec{NodeName: "nodeA"},
+				},
+			},
+			nodes: []*v1.Node{
+				{ObjectMeta: metav1.ObjectMeta{Name: "nodeA", Labels: map[string]string{"hostname": "nodeA"}}},
+			},
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "anti", Labels: map[string]string{"aaa": "", "bbb": ""}},
+				Spec: v1.PodSpec{
+					Affinity: &v1.Affinity{
+						PodAntiAffinity: &v1.PodAntiAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: newPodAffinityTerms([]string{"aaa", "ccc"}),
```

Comment:
Existing pod lable: `"aaa": "", "bbb": ""`
New pod AntiAffinity is `"aaa", "ccc"`

It seems not `violates all terms`? Or I mis-read sth?

Reply:
Thanks for the catch. It's a copy/paste typo... Fixed.